honeybee_3dm/config.py

```diff
@@ -0,0 +1,432 @@
+    )
+
+    include_child_layers: bool = Field(
+        False,
+        description='Boolean to indicate if objects from child layers are to be imported'
+    )
+
+    radiance_material: str = Field(
+        None,
+        description='Text string of the radiance identifier.'
+    )
+
+    grid_settings: GridSettings = Field(
+        None
+    )
+
+    honeybee_face_object: FaceObjects = Field(
+        None,
+        description='A text string for FaceObject to create either a Honeybee Shade,'
+        ' a Honeybee door, or a Honeybee aperture object.'
+    )
+
+    honeybee_face_type: FaceType = Field(
+        None,
+        description='A text string for FaceType to create either a Honeybee Wall,'
+        ' a Honeybee Floor, or a Honeybee aperture roof/ceiling, or a Honeybee.'
+        ' Floor.'
+    )
+
+
+class Config(BaseModel):
+    """Config for the config file."""
+
+    sources: Dict[str, str] = Field(
+        None,
+        description='Path to the radiance .mat file.'
+    )
+
+    layers: Dict[str, LayerConfig] = Field(
+        description='names of layer in rhino.'
+    )
+
+    @validator('sources')
+    def check_sources(cls, v):
+        sources = list(v.keys())
+        if len(sources) != 1:
+            raise ValueError('sources can only have one key.')
+
+        if sources[0] != 'radiance_material':
+            raise ValueError(
+                    f'invalid sources key: {sources[0]}.'
+                    'key must be radiance_material'
+            )
+        return v
+    
+    @validator('layers')
+    def check_layers(cls, v):
+        layers = list(v.keys())
+        if not layers:
+            raise ValueError('The config file cannot be used without layer names.')
+        return v
+
+
+def read_json(config_path):
+    """Get a dictionary from a config.json file.
+
+    Args:
+        path: A text string for the path to the config.json.
+
+    Returns:
+        A dictionary for the config.json
+    """
+    try:
+        with open(config_path) as fh:
+            config = json.load(fh)
+    except json.decoder.JSONDecodeError:
+        raise ValueError(
+            'Not a valid config file.'
+        )
+    return config
+
+
+def check_layers(file_3dm, config):
+    """Checks if layers in the config file are layers from the rhino file.
+
+    Args:
+        file_3dm: A rhino3dm file object.
+        config: Config dictionary.
+
+    Raises:
+        KeyError: If any of the layer names mentioned in the config file is not found
+            in the layers in the rhino file.
+
+    Returns:
+        Boolean value of True if no error is raised.
+    """
+    layers = [layer.Name for layer in file_3dm.Layers]
+    layer_check = [True for layer in config['layers'] if layer in layers]
+    if len(config['layers']) != layer_check.count(True):
+        raise KeyError(
+            'Only layer names from the Rhino file are allowed in the config file.'
+        )
+    else:
+        return True
```

Comment:
Am I right that you are trying to make sure all the layers in config file are also in Rhino? In that case can't we just check that?

```suggestion
    layer_check = [layer for layer in config['layers'] if layer not in layers]
    if layer_check:
        raise KeyError(
            'Only layer names from the Rhino file are allowed in the config file. '
            f'Found invalid layer names: {layer_check}'
        )
    return True
```

Reply:
Also what about child layers? Should we support `parent_layer::child_layer' for layer names?

Reply:
Each layer in the config file is checked against the list of all the layers in rhino. This list includes child layers as well.